src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefDataIntegrationTest.java

```diff
@@ -43,6 +43,64 @@ void should_retrieveAllServiceSkills_return_status_code_200()
         String role = "staff-admin";
 
 
+        final var staffWorkerSkillResponse = (StaffWorkerSkillResponse) caseworkerReferenceDataClient
+                .retrieveAllServiceSkills(StaffWorkerSkillResponse.class, path, role);
+
+        assertThat(staffWorkerSkillResponse).isNotNull();
+
+        List<ServiceSkill> serviceSkills = staffWorkerSkillResponse.getServiceSkills();
+
+        assertThat(serviceSkills).isNotNull();
+
+        ServiceSkill serviceSkill = serviceSkills.get(0);
+
+        assertThat(serviceSkill.getId()).isEqualTo("AAA7");
+
+        SkillDTO skillDTO = serviceSkill.getSkills().get(0);
+
+        assertThat(skillDTO.getSkillId()).isEqualTo(9L);
+        assertThat(skillDTO.getSkillCode()).isEqualTo("SKILL:AAA7:TEST1");
+        assertThat(skillDTO.getDescription()).isEqualTo("testskill1");
+        assertThat(skillDTO.getUserType()).isEqualTo("CTSC");
+    }
+
+    @Test
+    void should_retrieveAllServiceSkills_return_status_code_200_when_empty_servicecode()
+            throws JsonProcessingException {
+        String path = "/skill?service_codes=";
+
+        String role = "staff-admin";
+
+
+        final var staffWorkerSkillResponse = (StaffWorkerSkillResponse) caseworkerReferenceDataClient
+                .retrieveAllServiceSkills(StaffWorkerSkillResponse.class, path, role);
+
+        assertThat(staffWorkerSkillResponse).isNotNull();
+
+        List<ServiceSkill> serviceSkills = staffWorkerSkillResponse.getServiceSkills();
+
+        assertThat(serviceSkills).isNotNull();
+
+        ServiceSkill serviceSkill = serviceSkills.get(0);
+
+        assertThat(serviceSkill.getId()).isEqualTo("AAA7");
+
+        SkillDTO skillDTO = serviceSkill.getSkills().get(0);
+
+        assertThat(skillDTO.getSkillId()).isEqualTo(9L);
+        assertThat(skillDTO.getSkillCode()).isEqualTo("SKILL:AAA7:TEST1");
+        assertThat(skillDTO.getDescription()).isEqualTo("testskill1");
+        assertThat(skillDTO.getUserType()).isEqualTo("CTSC");
+    }
+
+    @Test
+    void should_retrieveAllServiceSkills_return_status_code_200_when_provide_servicecode()
+            throws JsonProcessingException {
+        String path = "/skill?service_codes=AAA7";
+
+        String role = "staff-admin";
+
+
```

Comment:
add test cases to cover below scenarios:
1. query for one service code where there is one service code matched out of 5 service codes and assert the expected count and skills data
2.query for two service codes where there are 2 service codes matched out of 5 service codes and assert the expected count and skills data

Reply:
As service code data is added through flyway script and added test cases to search on Service codes

Reply:
i don't see query for one service code test case

Reply:
done